### ISSUE TITLE
Federated learning

### ISSUE THREAD

**Issue by ymoisan** [NRCan/geo-deep-learning#5]
"In a typical machine learning system, an optimization algorithm like Stochastic Gradient Descent (SGD) runs on a large dataset partitioned homogeneously across servers in the cloud. Such highly iterative algorithms require low-latency, high-throughput connections to the training data. But in the Federated Learning setting, the data is distributed across millions of devices in a highly uneven fashion. In addition, these devices have significantly higher-latency, lower-throughput connections and are only intermittently available for training." 
https://ai.googleblog.com/2017/04/federated-learning-collaborative.html

Could this be applied in a later stage, e.g. to optimize our model through time ?

**Comment by mpelchat04** [comment]
Refer to [scratchpad](#60)
